### PR TITLE
feat: eventshub sender can send to resources in other namespaces

### PR DESCRIPTION
**Pull request by Cali0707**
# Changes

- Add a new k8s NamespacedAddress function to return an address in a different namespace than env
- Add some eventshub options to send to resources in other namespaces

**Release Note**

```release-note
Eventshub senders can deliver to resources in other namespaces
```

### CODE REVIEW

**Comment by Cali0707** [comment]
/cc @creydr @pierDipi @yijie-04 

This should help with fixing the cross namespace e2e test PR in eventing core

File: pkg/eventshub/options.go [hunk 1]
```diff
@@ -119,6 +139,28 @@ func StartSenderToResourceTLS(gvr schema.GroupVersionResource, name string, caCe
 	}
 }
 
+// StartSenderToNamespacedResourceTLS starts the sender in the eventshub pointing to the provided namespaced resource.
+// `caCerts` parameter is optional, if nil, it will fall back to use the addressable CA certs.
+// This can be used together with InputEvent, AddTracing, EnableIncrementalId, InputEncoding and InputHeader options
+func StartSenderToNamespacedResourceTLS(gvr schema.GroupVersionResource, name, namespace string, caCerts *string) EventsHubOption {
```
Comment (pierDipi) [inline pkg/eventshub/options.go:145]: Similar question for this, we can call this in `StartSenderToResourceTLS`

**Review by creydr — COMMENTED** [review]
/lgtm

**Pull request merged**